### DISCUSSION TITLE
i found that when i used  'resampling=bilinear' for cog,  The color is not displayed properly

### DISCUSSION THREAD

**Discussion by zqhay** [developmentseed/titiler#514] [Q&A]
HI 
I used custom colormap and  resampling is  bilinear . The TIF ranges from -3 to 33 and I use the rescale from 0 to 30.Then  a strange thing happened.  Where the ocean meets the land, TIF values are abrupt, There's no value from 0 to 15, but it showed colors for costom colormap.
This is my display png, as you can see, The border should not appear blue,

![Snipaste_2022-09-19_16-48-14](https://user-images.githubusercontent.com/49359535/190981690-30ffeb8d-cfd4-48bb-a849-abdea60c0dcf.png)

This is my colormap:
`[(0, '#FFFFFF01'), (0.17, '#8200FF01'), (0.33, '#004DFF01'), (0.5, '#00B0FF01'), (0.67, '#BDFF0001'), (0.83, '#FFE50001'), (1, '#FF100001')]`

This is my request:
![image](https://user-images.githubusercontent.com/49359535/190983989-c92527d9-dea7-4cb4-91d2-15e3b72a0ce8.png)

I think the rescale parameter is a mapping to the TIF value. Am I wrong?

Can you give me  some suggestions? what am I doing wrong? I just want to show the value of the TIF range through the colormap
Thanks.

**Comment by vincentsarago** [comment]
is the tiff nodata value encoded in a nodata value, an alpha mask or a internal mask? 

could you share the data url? (I don't really want to transcript the url on the image 😬

**Reply by zqhay** [reply]
Sorry. I'm just trying to describe the problem I'm having.
The tiff nodata value is  ma.masked,  and  I do not  know  set what for  the nodata param.
that's the data url  :
`http://192.168.152.128:8020/cog/tiles/4/14/4.png?scale=1&TileMatrixSetId=WebMercatorQuad&url=http://192.168.31.114:30966/ocean-data/data/16d4c911-336b-405f-87d3-66789a41ee68/990100c9-647a-491a-90b8-f4f38993a0a9/bde49d8c-6bd5-431d-961a-f059909de2d4/201901/cog/_D5.0.tif&bidx=1&unscale=true&resampling=bilinear&return_mask=true&colormap={"0":[255,255,255,0],"1":[252,249,255,255],"2":[249,243,255,255],"3":[246,237,255,255],"4":[243,231,255,255],"5":[240,225,255,255],"6":[237,219,255,255],"7":[234,213,255,255],"8":[231,207,255,255],"9":[229,202,255,255],"10":[226,196,255,255],"11":[223,190,255,255],"12":[220,184,255,255],"13":[217,178,255,255],"14":[214,172,255,255],"15":[211,166,255,255],"16":[208,160,255,255],"17":[205,155,255,255],"18":[203,149,255,255],"19":[200,143,255,255],"20":[197,137,255,255],"21":[194,131,255,255],"22":[191,125,255,255],"23":[188,119,255,255],"24":[185,113,255,255],"25":[182,107,255,255],"26":[180,102,255,255],"27":[177,96,255,255],"28":[174,90,255,255],"29":[171,84,255,255],"30":[168,78,255,255],"31":[165,72,255,255],"32":[162,66,255,255],"33":[159,60,255,255],"34":[156,55,255,255],"35":[154,49,255,255],"36":[151,43,255,255],"37":[148,37,255,255],"38":[145,31,255,255],"39":[142,25,255,255],"40":[139,19,255,255],"41":[136,13,255,255],"42":[133,7,255,255],"43":[131,2,255,255],"44":[127,1,255,255],"45":[124,3,255,255],"46":[121,5,255,255],"47":[118,6,255,255],"48":[115,8,255,255],"49":[111,10,255,255],"50":[108,12,255,255],"51":[105,14,255,255],"52":[102,16,255,255],"53":[99,18,255,255],"54":[96,20,255,255],"55":[92,21,255,255],"56":[89,23,255,255],"57":[86,25,255,255],"58":[83,27,255,255],"59":[80,29,255,255],"60":[76,31,255,255],"61":[73,33,255,255],"62":[70,35,255,255],"63":[67,37,255,255],"64":[64,38,255,255],"65":[61,40,255,255],"66":[57,42,255,255],"67":[54,44,255,255],"68":[51,46,255,255],"69":[48,48,255,255],"70":[45,50,255,255],"71":[41,52,255,255],"72":[38,54,255,255],"73":[35,55,255,255],"74":[32,57,255,255],"75":[29,59,255,255],"76":[25,61,255,255],"77":[22,63,255,255],"78":[19,65,255,255],"79":[16,67,255,255],"80":[13,69,255,255],"81":[10,71,255,255],"82":[6,72,255,255],"83":[3,74,255,255],"84":[0,76,255,255],"85":[0,78,255,255],"86":[0,81,255,255],"87":[0,83,255,255],"88":[0,85,255,255],"89":[0,88,255,255],"90":[0,90,255,255],"91":[0,92,255,255],"92":[0,94,255,255],"93":[0,97,255,255],"94":[0,99,255,255],"95":[0,101,255,255],"96":[0,104,255,255],"97":[0,106,255,255],"98":[0,108,255,255],"99":[0,110,255,255],"100":[0,113,255,255],"101":[0,115,255,255],"102":[0,117,255,255],"103":[0,120,255,255],"104":[0,122,255,255],"105":[0,124,255,255],"106":[0,126,255,255],"107":[0,129,255,255],"108":[0,131,255,255],"109":[0,133,255,255],"110":[0,136,255,255],"111":[0,138,255,255],"112":[0,140,255,255],"113":[0,142,255,255],"114":[0,145,255,255],"115":[0,147,255,255],"116":[0,149,255,255],"117":[0,152,255,255],"118":[0,154,255,255],"119":[0,156,255,255],"120":[0,158,255,255],"121":[0,161,255,255],"122":[0,163,255,255],"123":[0,165,255,255],"124":[0,168,255,255],"125":[0,170,255,255],"126":[0,172,255,255],"127":[0,174,255,255],"128":[2,176,252,255],"129":[6,178,246,255],"130":[10,180,240,255],"131":[15,182,234,255],"132":[19,184,228,255],"133":[23,186,222,255],"134":[28,187,216,255],"135":[32,189,210,255],"136":[37,191,205,255],"137":[41,193,199,255],"138":[45,195,193,255],"139":[50,196,187,255],"140":[54,198,181,255],"141":[58,200,175,255],"142":[63,202,169,255],"143":[67,204,163,255],"144":[71,206,157,255],"145":[76,207,152,255],"146":[80,209,146,255],"147":[85,211,140,255],"148":[89,213,134,255],"149":[93,215,128,255],"150":[98,217,122,255],"151":[102,218,116,255],"152":[106,220,110,255],"153":[111,222,105,255],"154":[115,224,99,255],"155":[119,226,93,255],"156":[124,227,87,255],"157":[128,229,81,255],"158":[132,231,75,255],"159":[137,233,69,255],"160":[141,235,63,255],"161":[146,237,57,255],"162":[150,238,52,255],"163":[154,240,46,255],"164":[159,242,40,255],"165":[163,244,34,255],"166":[167,246,28,255],"167":[172,247,22,255],"168":[176,249,16,255],"169":[180,251,10,255],"170":[185,253,5,255],"171":[189,254,0,255],"172":[190,254,0,255],"173":[192,253,0,255],"174":[194,252,0,255],"175":[195,252,0,255],"176":[197,251,0,255],"177":[198,251,0,255],"178":[200,250,0,255],"179":[202,249,0,255],"180":[203,249,0,255],"181":[205,248,0,255],"182":[207,247,0,255],"183":[208,247,0,255],"184":[210,246,0,255],"185":[211,245,0,255],"186":[213,245,0,255],"187":[215,244,0,255],"188":[216,244,0,255],"189":[218,243,0,255],"190":[219,242,0,255],"191":[221,242,0,255],"192":[223,241,0,255],"193":[224,240,0,255],"194":[226,240,0,255],"195":[228,239,0,255],"196":[229,238,0,255],"197":[231,238,0,255],"198":[232,237,0,255],"199":[234,237,0,255],"200":[236,236,0,255],"201":[237,235,0,255],"202":[239,235,0,255],"203":[241,234,0,255],"204":[242,233,0,255],"205":[244,233,0,255],"206":[245,232,0,255],"207":[247,231,0,255],"208":[249,231,0,255],"209":[250,230,0,255],"210":[252,230,0,255],"211":[253,229,0,255],"212":[255,227,0,255],"213":[255,222,0,255],"214":[255,217,0,255],"215":[255,212,0,255],"216":[255,207,0,255],"217":[255,202,0,255],"218":[255,197,0,255],"219":[255,192,0,255],"220":[255,187,0,255],"221":[255,183,0,255],"222":[255,178,0,255],"223":[255,173,0,255],"224":[255,168,0,255],"225":[255,163,0,255],"226":[255,158,0,255],"227":[255,153,0,255],"228":[255,148,0,255],"229":[255,143,0,255],"230":[255,138,0,255],"231":[255,133,0,255],"232":[255,129,0,255],"233":[255,124,0,255],"234":[255,119,0,255],"235":[255,114,0,255],"236":[255,109,0,255],"237":[255,104,0,255],"238":[255,99,0,255],"239":[255,94,0,255],"240":[255,89,0,255],"241":[255,84,0,255],"242":[255,79,0,255],"243":[255,74,0,255],"244":[255,70,0,255],"245":[255,65,0,255],"246":[255,60,0,255],"247":[255,55,0,255],"248":[255,50,0,255],"249":[255,45,0,255],"250":[255,40,0,255],"251":[255,35,0,255],"252":[255,30,0,255],"253":[255,25,0,255],"254":[255,20,0,255],"255":[255,16,0,255]}&rescale=0,30`


[that's my tif](https://github.com/fallingwenwen/problems/blob/main/titiler/colormap.txt)
[that's the colormap .txt  for qgis](https://github.com/fallingwenwen/problems/blob/main/titiler/_D5.0.tif), I want to get the effect of QGIS，Could you tell me what to do
![image](https://user-images.githubusercontent.com/49359535/191159601-f55e306a-7121-4b2a-8d82-448e8df2917c.png)
